### DISCUSSION TITLE
How to customize UserMessage View in a Group Channel?

### DISCUSSION THREAD

**Discussion by lookdeceline** [sendbird/sendbird-swiftui-ios#12] [1. Chatting in a channel]
![Languages](https://img.shields.io/badge/SwiftUI-0078FC?logo=swift&logoColor=white) [![Version](https://img.shields.io/badge/SendbirdSwiftUI-1.0.0_beta.0-6210CC.svg)](https://github.com/sendbird/sendbird-uikit-ios/releases/tag/v1.0.0)

### Question
How to customize UserMessage View depending on the value of `message.customType`?

### Answer
You can use `listItem` view adaptor's `.userMessageView(content:)` method when initializing `GroupChannelView`.
The `.userMessageView(content:)` method allows you to customize the User Message View. 

```swift
GroupChannelView(
      channelURL: {CHANNEL_URL},
      listItem: {
          .init()
          .userMessageView { config in
              // Your custom logic goes here.
              if config.message.customType == "target_string" {
                  // View 
              } else {
                  // View
              }
          }
      }
  )
```

**Comment by luchi2300** [comment]
Hello @lookdeceline, Is there a way to hide the username?

![Screenshot 2025-06-13 at 15 36 41](https://github.com/user-attachments/assets/5d82d9ce-a1d6-4cb3-a9a0-58fa21071461)
